contracts/yieldaggregator-bonus/src/execute/register_bonus_window.rs

```diff
@@ -20,17 +22,33 @@ pub fn execute_register_bonus_window(
         return Err(ContractError::Unauthorized {});
     }
 
-    let id = 0u64;
+    let deposit = one_coin(&info)?;
+    if deposit.denom != msg.denom {
+        return Err(ContractError::NoAllowedToken {});
+    }
+
+    if deposit.amount < msg.budget_for_all {
```

Comment:
Is there any specific reason keeping deposit.denom, deposit.amount, msg.denom, msg.budget_for_all ? I think only deposit could be used and denom and budget_for_all could be removed on msg struct